data-plane/gateway/controller/src/service.rs

```diff
@@ -190,9 +192,57 @@ impl ControllerService {
                             eprintln!("failed to send ACK: {}", e);
                         }
                     }
+                    crate::api::proto::api::v1::control_message::Payload::SubscriptionListRequest(_) => {
+                        let mut batch = Vec::new();
+                        const BATCH_SIZE: usize = 100;
+
+                        self.message_processor.subscription_table().for_each(
+                            |agent_type, agent_id, local, remote| {
+                                batch.push(SubscriptionEntry {
+                                    company: agent_type.organization_name(),
+                                    namespace: agent_type.namespace_name(),
+                                    agent_name: agent_type.agent_type_name(),
+                                    agent_id: Some(agent_id),
+                                    local_connection_ids: local.to_vec(),
+                                    remote_connection_ids: remote.to_vec(),
+                                });
+
+                                if batch.len() == BATCH_SIZE {
+                                    let entries_to_send = std::mem::take(&mut batch);
+                                    let resp = ControlMessage {
+                                        message_id: uuid::Uuid::new_v4().to_string(),
+                                        payload: Some(
+                                            crate::api::proto::api::v1::control_message::Payload::SubscriptionListResponse(
+                                                SubscriptionListResponse { entries: entries_to_send }
+                                            ),
+                                        ),
+                                    };
+
+                                    let _ = tx.try_send(Ok(resp));
```

Comment:
hmm, let's try not to send anything while keeping the table lock. Probably better filling the vec and the use something like [chunks](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks) to operate on it. What do you think?